KlimaDAO/klimadao
# Makka/base poc

## Description

This PR allows retirement of BCT on Base using Axelar.

## Related Ticket

Closes #2334

## How to Test

Test URL: https://base-git-makka-base-poc-klimadao.vercel.app/

1. Bridge some Klima over to Base on Squid Router - https://v2.app.squidrouter.com/?chains=137%2C8453&tokens=0x4e78011Ce80ee02d2c3e649Fb657E45898257815%2C0xDCEFd8C8fCc492630B943ABcaB3429F12Ea9Fea2
2. Ensure you have some ETH on base for gas fees.
3. Navigate to the base landing page, enter the details in the required fields including the quantity. 
4. Happy path - Click "Retire Carbon" and attempt to execute a retirement, if successful - you should receive a "Transaction complete" message with a link to the transaction on Axelar where the progress can be tracked.
5. Once the transaction is complete (generally can take up to 30 minutes). Verify the retirement was successful with the correct data - should be visible under your [wallet address on carbonmark - [example](https://www.carbonmark.com/retirements/0x6cb03a22699e50c8ce4fd18cd528edfa8cfe7ed2).
6. Error handling - Ensure the error messages on required fields are valid (i.e. insufficient balance, no quantity added, empty retirement message, empty beneficiary string, wrong network (anything other than base currently), logged out user etc...
7. Further Error handling - Ensure appropriate messages are shown in the transaction modal when a user for example rejects the transaction.
